src/utils.js

```diff
@@ -239,3 +241,60 @@ export const validateDropDownValue = (shouldValidate) => (value) => {
 
   return undefined;
 };
+
+export const getUserPreferences = (mediatedRequest, userData, servicePoints) => {
+  const userPreferences = {};
+
+  if (mediatedRequest) {
```

Comment:
Just a note: To avoid nesting of "if," early return could be used. From a readability perspective, it could be better.

Reply:
We usually trying to use positive scenarios at the beginning `if (someCondition)`
instead of using negative `if (!someCondition)`.
If you do not mind I would leave it as is.
But I agree with you that your suggestion will have positive impact on readability.